pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/AvgShedder.java

```diff
@@ -0,0 +1,306 @@
+                log.warn("Broker {} is overloaded despite having no bundles", pair.getRight());
+            }
+
+            // do shedding
+            log.info(
+                    "broker:[{}] is planning to shed bundles to broker:[{}]. "
+                            + "maxBroker stat:scores:{}, throughput:{}, msgRate:{}. "
+                            + "minBroker stat:scores:{}, throughput:{}, msgRate:{}. "
+                            + "isMsgRateToOffload:{},  trafficMarkedToOffload:{}",
+                    overloadedBroker, underloadedBroker, brokerScoreMap.get(overloadedBroker), maxThroughput,
+                    maxMsgRate, brokerScoreMap.get(underloadedBroker), minThroughput, minMsgRate,
+                    isMsgRateToOffload, trafficMarkedToOffload);
+
+            loadData.getBundleDataForLoadShedding().entrySet().stream().filter(e ->
+                    maxLocalBrokerData.getBundles().contains(e.getKey())
+            ).filter(e ->
+                    !recentlyUnloadedBundles.containsKey(e.getKey())
+            ).map((e) -> {
+                BundleData bundleData = e.getValue();
+                TimeAverageMessageData shortTermData = bundleData.getShortTermData();
+                double traffic = isMsgRateToOffload
+                        ? shortTermData.getMsgRateIn() + shortTermData.getMsgRateOut()
+                        : shortTermData.getMsgThroughputIn() + shortTermData.getMsgThroughputOut();
+                return Pair.of(e, traffic);
+            }).sorted((e1, e2) ->
+                    Double.compare(e2.getRight(), e1.getRight())
+            ).forEach(e -> {
+                Map.Entry<String, BundleData> bundle = e.getLeft();
+                double traffic = e.getRight();
+                if (traffic > 0 && traffic <= trafficMarkedToOffload.getValue()) {
+                    selectedBundlesCache.put(overloadedBroker, bundle.getKey());
+                    bundleBrokerMap.put(bundle.getValue(), underloadedBroker);
+                    trafficMarkedToOffload.add(-traffic);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Found bundle to unload:{}, isMsgRateToOffload:{}, traffic:{}",
+                                bundle, isMsgRateToOffload, traffic);
+                    }
+                }
+            });
+        }
+        return selectedBundlesCache;
+    }
+
+    @Override
+    public void onActiveBrokersChange(Set<String> activeBrokers) {
+        LoadSheddingStrategy.super.onActiveBrokersChange(activeBrokers);
+    }
+
+    private Double calculateScores(LocalBrokerData localBrokerData, final ServiceConfiguration conf) {
+        return localBrokerData.getMaxResourceUsageWithWeight(
+                conf.getLoadBalancerCPUResourceWeight(),
+                conf.getLoadBalancerDirectMemoryResourceWeight(),
+                conf.getLoadBalancerBandwithInResourceWeight(),
+                conf.getLoadBalancerBandwithOutResourceWeight()) * 100;
```

Comment:
These configs are deprecated, use the correctly spelled word `Bandwidth` instead of `Bandwith`.